core-bundle/src/DataContainer/DataContainerOperationsBuilder.php

```diff
@@ -0,0 +1,282 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\DataContainer;
+
+use Contao\Backend;
+use Contao\CoreBundle\Security\ContaoCorePermissions;
+use Contao\DataContainer;
+use Contao\Image;
+use Contao\Input;
+use Contao\StringUtil;
+use Contao\System;
+use Symfony\Bundle\SecurityBundle\Security;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Twig\Environment;
+
+/**
+ * @internal
+ */
+class DataContainerOperationsBuilder
+{
+    public function __construct(
+        private readonly Environment $twig,
+        private readonly Security $security,
+        private readonly UrlGeneratorInterface $urlGenerator,
+    ) {
+    }
+
+    public function generateButtons(string $table, array $record, DataContainer $dataContainer, callable|null $legacyCallback = null): \Stringable|string
+    {
+        if (!\is_array($GLOBALS['TL_DCA'][$table]['list']['operations'] ?? null)) {
+            return '';
+        }
+
+        $operations = [];
+
+        foreach ($GLOBALS['TL_DCA'][$table]['list']['operations'] as $k => $v) {
+            $v = \is_array($v) ? $v : [$v];
+            $operation = $this->generateOperation($k, $v, $table, $record, $dataContainer, $legacyCallback);
+
+            if ($operation) {
+                $operations[] = $operation;
+            }
+        }
+
+        return $this->createBuilder($operations);
+    }
+
+    public function generateHeaderButtons(string $table, array $record, DataContainer $dataContainer, callable|null $legacyCallback = null): \Stringable|string
+    {
+        if (!\is_array($GLOBALS['TL_DCA'][$table]['list']['operations'] ?? null)) {
+            return '';
+        }
+
+        $operations = [];
+
+        foreach ($GLOBALS['TL_DCA'][$table]['list']['operations'] as $k => $v) {
+            // Edit operation should be shown in header by default
+            $showInHeader = ($v['showInHeader'] ?? false) || ('edit' === $k && false !== ($v['showInHeader'] ?? true));
+
+            if (!$showInHeader || ('select' === Input::get('act') && !($v['showOnSelect'] ?? null))) {
```

Comment:
```suggestion
            // Show edit operation in the header by default (backwards compatibility)
            if ('edit' === $k && !isset($v['showInHeader'])) {
                $v['showInHeader'] = true;
            }

            if (!$v['showInHeader'] || ('select' === Input::get('act') && !($v['showOnSelect'] ?? null))) {
```

This way we won‘t forget to remove the if condition in Contao 6.